addons/l10n_sa_edi/wizard/l10n_sa_edi_otp_wizard.py

```diff
@@ -0,0 +1,18 @@
+from odoo import fields, models, _
+from odoo.exceptions import UserError
+
+
+class RequestZATCAOtp(models.TransientModel):
+    _name = 'l10n_sa_edi.otp.wizard'
+    _description = 'Request ZATCA OTP'
+
+    l10n_sa_otp = fields.Char("OTP", copy=False, help="OTP required to get a CCSID. Can only be acquired through "
+                                                      "the Fatoora portal.")
+
+    def validate(self):
+        if not self.l10n_sa_otp:
+            raise UserError(_("You need to provide an OTP to be able to request a CCSID"))
+        journal_id = self.env['account.journal'].browse(self.env.context.get('active_id'))
+        if self.env.context.get('renew_pcsid'):
```

Comment:
and a renew pcsid field?

Reply:
I'm not sure what you mean. The wizard can be used to set the OTP in one of two cases: When we try to get the CCSID, or when we try to renew the PCSID. In the latter case, we just pass a key in the context to know that this is to renew the PCSID and use the appropriate API. Do you mean to add a boolean field on the wizard and set its value through the context ?

Reply:
Yes.  Before, sometimes the context was lost along the way and so you needed to do it anyways.  And it might be in handy to change it in the wizard.